server/controllers/payroll/multiplePayroll/index.js

```diff
@@ -0,0 +1,1177 @@
+                params : [allRubrics],
+              });
+            }
+
+            if (holidaysElements.length) {
+              allTransactions.push({
+                query : setHolidayPaiement,
+                params : [holidaysElements],
+              });
+            }
+
+            if (offDaysElements.length) {
+              allTransactions.push({
+                query : setOffDayPaiement,
+                params : [offDaysElements],
+              });
+            }
+
+            return allTransactions;
+          });
+      }))
+        .then((results) => {
+          const postingJournal = db.transaction();
+
+          results.forEach(transac => {
+            transac.forEach(item => {
+              postingJournal.addQuery(item.query, item.params);
+            });
+          });
+
+          return postingJournal.execute();
+        })
+        .then(() => {
+          res.sendStatus(201);
+        });
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+ * POST /multiple_payroll/:id/configuration
+ */
+function setConfiguration(req, res, next) {
+  const data = req.body.data;
+  const transaction = db.transaction();
+  const currencyId = req.session.enterprise.currency_id;
+  const enterpriseId = req.session.enterprise.id;
+
+  // If tax IPR is not defined Else Use Currency ID
+  const iprCurrencyId = data.iprScales.length ? data.iprScales[0].currency_id : currencyId;
+
+  const iprScales = data.iprScales;
+  const employee = data.employee;
+  const payrollConfigurationId = req.params.id;
+  const paiementUuid = uuid();
+
+  const uid = db.bid(paiementUuid);
+
+  const allRubrics = [];
+  const holidaysElements = [];
+  const offDaysElements = [];
+
+  let enterpriseExchangeRate = 0;
+  let iprExchangeRate = 0;
+
+  Exchange.getExchangeRate(enterpriseId, data.currency_id, new Date())
+    .then((exchange) => {
+      enterpriseExchangeRate = currencyId === data.currency_id ? 1 : exchange.rate;
+
+      return Exchange.getExchangeRate(enterpriseId, iprCurrencyId, new Date());
+    })
+    .then((exchangeIpr) => {
+      iprExchangeRate = exchangeIpr.rate;
+      const DECIMAL_PRECISION = 2;
+
+      let offDaysCost = 0;
+      let holidaysCost = 0;
+
+      // Calcul Daily Salary
+      const totalDayPeriod = data.daysPeriod.working_day;
+
+      const dailySalary = employee.individual_salary ?
+        (employee.individual_salary / totalDayPeriod) :
+        (employee.basic_salary / totalDayPeriod);
+
+      const workingDayCost = dailySalary * data.working_day;
+      const nbChildren = employee.nb_enfant;
+
+      /**
+     * Some institution allocates a percentage for the offday and holiday payment,
+     * the calculation of this rate is found by calculating the equivalence of the daily wage with
+     * the percentage of the offday or holiday.
+    */
+
+      data.offDays.forEach(offDay => {
+        const offDayValue = ((dailySalary * offDay.percent_pay) / 100);
+        offDaysCost += offDayValue;
```

Comment:
I'm pretty worried about doing lots of math in JavaScript, but I'm not sure of a better solution.  It would be nice to do as much of this in SQL as possible.  Maybe we can migrate it into a stored procedure eventually.
```js
console.log(1.3 + 0.4(
// => 1.7000000000000002
```